internal/otelcollector/config/metric/gateway/processors.go

```diff
@@ -79,6 +79,19 @@ func makeDropIfInputSourceOtlpConfig() *FilterProcessor {
 	}
 }
 
+func makeDropDiagnosticMetricsForInput(inputSourceCondition string) *FilterProcessor {
```

Comment:
can we move it to a separate file like this: https://github.com/kyma-project/telemetry-manager/blob/main/internal/otelcollector/config/metric/agent/drop_internal_communication.go and have a test ?

Reply:
separate unit test added